### PR TITLE
ja: cr2 Chromium translation

### PR DESCRIPTION
**Pull request by henrif75**
Professional translations for CR2 content.
Changes are being staged in a separate branch until all Japanese content is reviewed.

#1463 #652

### CODE REVIEW

**Comment by kantasv** [comment]
Hi, @henrif75 . Thanks for raising this and I'll try to take a look into this :)

**Comment by henrif75** [comment]
> Hi, @henrif75 . Thanks for raising this and I'll try to take a look into this :)

Thanks @kantasv!

**Comment by kantasv** [comment]
Hi Henri and JA translation team. Sorry, I don't have bandwidth to support this. Could someone take this?

File: po/ja.po [hunk 1]
```diff
@@ -12449,44 +12463,62 @@ msgid ""
...
+"ツールチェーンとクレートが使用されます。"
 
 #: src/chromium/cargo.md
 msgid ""
 "Use `cargo`, but [restrict yourself to Chromium's audited toolchain and "
 "crates](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/"
 "docs/rust.md#Using-cargo)"
 msgstr ""
+"`cargo` を使用しますが、[Chromium の監査済みのツールチェーンとクレートに制限"
+"してください](https://chromium.googlesource.com/chromium/src/+/refs/heads/"
```
Comment (HidenoriKobayashi) [inline po/ja.po:12501]:
```suggestion
"`cargo` を使用しますが、実際の利用を[Chromium の監査済みのツールチェーンとクレートに制限"
"します](https://chromium.googlesource.com/chromium/src/+/refs/heads/"
```

File: po/ja.po [hunk 2]
```diff
@@ -12449,44 +12463,62 @@ msgid ""
...
 #: src/chromium/cargo.md
 msgid ""
 "From here on we'll be focusing on `gn` and `ninja`, because this is how Rust "
 "code can be built into the Chromium browser. At the same time, Cargo is an "
 "important part of the Rust ecosystem and you should keep it in your toolbox."
 msgstr ""
+"ここからは、`gn` と `ninja` に焦点を当てます。これらを使用することで、"
+"Chromium ブラウザに Rust コードを組み込むことができます。同時に、Cargo は "
+"Rust エコシステムの重要な部分であるため、使いこなせるようになっているべきだか"
+"らです。"
```
Comment (HidenoriKobayashi) [inline po/ja.po:12521]:
```suggestion
"Chromium ブラウザに Rust コードを組み込むことができます。それとは別に、Cargo は "
"Rust エコシステムの重要な部分であり、使いこなせるようになっているべきです。"
""
```

File: po/ja.po [hunk 3]
```diff
@@ -12495,32 +12527,40 @@ msgstr "練習問題"
...
+"ラリ、人々を信頼しなければならないかについて話し合います。"
 
 #: src/chromium/cargo.md
 msgid ""
 "Ask students to avoid peeking at the speaker notes before completing the "
 "exercise. Assuming folks taking the course are physically together, ask them "
 "to discuss in small groups of 3-4 people."
 msgstr ""
+"受講者に、演習を完了する前にスピーカー ノートをのぞかないようお願いしてくださ"
+"い。コースの受講者同士が物理的に集まっていると仮定して、3～4 人の少人数のグ"
```
Comment (HidenoriKobayashi) [inline po/ja.po:12555]:
```suggestion
"い。コースの受講者同士が地理的に集まっていると仮定して、3～4 人の少人数のグ"
```

File: po/ja.po [hunk 4]
```diff
@@ -12530,18 +12570,27 @@ msgid ""
 "(`clap` for command-line parsing, `serde` for serializing/deserializing to/"
 "from various formats, `itertools` for working with iterators, etc.)."
 msgstr ""
+"ツールの作成時や Chromium の一部のプロトタイピング時に、crates.io ライブラリ"
+"の充実したエコシステムにアクセスできるのは素晴らしいことです。ほぼすべてにつ"
```
Comment (HidenoriKobayashi) [inline po/ja.po:12574]:
```suggestion
"の充実したエコシステムにアクセスできるのは素晴らしいことです。ほぼすべての事柄につ"
```

File: po/ja.po [hunk 5]
```diff
@@ -12714,28 +12815,50 @@ msgid ""
 "-'\n"
 "```"
 msgstr ""
+"```bob\n"
```
Comment (HidenoriKobayashi) [inline po/ja.po:12818]: Could you add an appropriate number of spaces to shape the diagram?

File: po/ja.po [hunk 6]
```diff
@@ -12957,49 +13169,67 @@ msgid ""
 "and run Chromium, and ensure that \"Hello from Rust!\" is printed lots of "
 "times."
 msgstr ""
+"この関数を `ui/base/resource/resource_bundle.cc` 内のどこかから呼び出します"
+"が、`ResourceBundle::MaybeMangleLocalizedString` の先頭から呼び出すことをおす"
+"すめします。Chromium をビルドして実行し、\"Hello from Rust!\" が何度も出力さ"
```
Comment (HidenoriKobayashi) [inline po/ja.po:13174]:
```suggestion
"この関数を `ui/base/resource/resource_bundle.cc` 内のどこかから呼び出します。"
"おすすめは`ResourceBundle::MaybeMangleLocalizedString` の先頭から呼び出すこと"
"です。Chromium をビルドして実行し、\"Hello from Rust!\" が何度も出力さ"
```

File: po/ja.po [hunk 7]
```diff
@@ -13090,24 +13353,35 @@ msgid ""
...
 msgstr ""
+"Rust 関数を `gtest` テストケースとして使用する（`#[gtest(...)]` 属性を使"
+"用）。"
 
 #: src/chromium/testing/rust-gtest-interop.md
 msgid ""
 "Use `expect_eq!` and similar macros (similar to `assert_eq!` but not "
 "panicking and not terminating the test when the assertion fails)."
 msgstr ""
+"`expect_eq!` などのマクロを使用する（`assert_eq! と似ていますが、アサーション"
```
Comment (HidenoriKobayashi) [inline po/ja.po:13383]:
```suggestion
"`expect_eq!` などのマクロを使用する（`assert_eq!`と似ていますが、アサーション"
```

File: po/ja.po [hunk 8]
```diff
@@ -13593,6 +14017,13 @@ msgid ""
 "writing/the-cxx-debate), but strictly, bringing any foreign code into a Rust "
 "binary can cause unexpected behavior from Rust's perspective."
 msgstr ""
+"大まかに答えると、C/C++ コードは通常の Rust 標準では「安全」ではありません。"
+"Rust から C/C++ に行ったり来たりすると、メモリに対して任意の処理が行われ、"
+"Rust 独自のデータ レイアウトの安全性が損なわれる可能性があります。C/C++ の相"
+"互運用で `unsafe` キーワードが多すぎると、'unsafe'に対する注目度が薄れるの"
```
Comment (HidenoriKobayashi) [inline po/ja.po:14023]:
```suggestion
"互運用で `unsafe` キーワードが多すぎると、`unsafe`に対する注目度が薄れるの"
```

File: po/ja.po [hunk 9]
```diff
@@ -13770,76 +14242,80 @@ msgstr "ライブラリ"
...
 
 #: src/chromium/adding-third-party-crates.md
 msgid "Lots"
-msgstr ""
+msgstr "多数"
 
 #: src/chromium/adding-third-party-crates.md
 msgid "Consistent: `Cargo.toml`"
-msgstr ""
+msgstr "一貫性がある: `Cargo.toml`"
```
Comment (HidenoriKobayashi) [inline po/ja.po:14258]:
```suggestion
msgstr "一貫して `Cargo.toml`"
```

File: po/ja.po [hunk 10]
```diff
@@ -13770,76 +14242,80 @@ msgstr "ライブラリ"
...
 
 #: src/chromium/adding-third-party-crates.md
 msgid "How to audit its source code for sufficient safety."
-msgstr ""
+msgstr "クレートのソースコードの十分な安全性を監査する方法。"
 
 #: src/chromium/adding-third-party-crates/configuring-cargo-toml.md
 msgid "Configuring the `Cargo.toml` file to add crates"
-msgstr ""
+msgstr " `Cargo.toml` ファイルによりクレートを追加する方法。"
```
Comment (HidenoriKobayashi) [inline po/ja.po:14318]:
```suggestion
msgstr " `Cargo.toml` ファイルによりクレートを追加する方法"
```

File: po/ja.po [hunk 11]
```diff
@@ -14072,6 +14601,11 @@ msgid ""
 "`allow-first-party-usage=false`. There are several examples already in that "
 "file:"
 msgstr ""
+"作成している場合は、[`gnrt_config.toml`](../configure-gnrt-config-toml.md) を"
```
Comment (HidenoriKobayashi) [inline po/ja.po:14604]:
```suggestion
"もしファイルが作成されるようになっていたら、[`gnrt_config.toml`](../configure-gnrt-config-toml.md) を"
```

File: po/ja.po [hunk 12]
```diff
@@ -14139,6 +14695,12 @@ msgid ""
 "                     +------------+      +----------------------+\n"
 "```"
 msgstr ""
+"```bob\n"
```
Comment (HidenoriKobayashi) [inline po/ja.po:14698]: This diagram looks broken. Could you take a look please?

**Review by HidenoriKobayashi — COMMENTED** [review]
I've gone through the translation one more time, looking at the result with the browser. Please take a look at the comments.

File: po/ja.po [hunk 13]
```diff
@@ -12848,107 +13013,153 @@ msgid ""
...
 "164; the outline is in the file explorer pane in vscode; typical vim/CoC "
 "bindings = space o)"
 msgstr ""
+"**outline** のデモを行い、`QrCode::with_bits` メソッドに移動します（164 行目"
+"付近。アウトラインは VSCode のファイル エクスプローラ ペインにあります。一般"
+"的な vim/CoC バインディング = space o）。"
 
 #: src/chromium/build-rules/vscode.md
 msgid ""
 "Demo **type annotations** (there are quote a few nice examples in the "
```
[the remark this answers is not in the document]
Comment (henrif75) [reply po/ja.po:13118]: It could be, I'll need to fix the original file.

**Review by HidenoriKobayashi — APPROVED** [review]
LGTM. Thanks!

**Comment by henrif75** [comment]
> LGTM. Thanks!

@HidenoriKobayashi

**Pull request merged**